src/components/YourCompany/FolioInformation.vue

```diff
@@ -0,0 +1,77 @@
+<template>
+  <FolioNumber
+    :initialValue="getFolioNumber"
+    :originalValue="originalFolioNumber"
+    :editLabel="editLabel"
+    :editedLabel="editedLabel"
+    :disableActions="isCorrectionFiling"
+    :invalidSection="invalidSection"
+    @newFolioNumber="onNewFolioNumber($event)"
+    @haveChanges="onHaveChanges($event)"
+    @isEditing="onIsEditing($event)"
+  />
+</template>
+
+<script lang="ts">
+import { Component, Emit, Mixins, Prop, Vue } from 'vue-property-decorator'
+import { Action, Getter } from 'vuex-class'
+import { ActionBindingIF, IncorporationFilingIF } from '@/interfaces'
+import { AuthApiMixin, CommonMixin } from '@/mixins'
+// TODO: move FolioNumber to bcrs-shared-components (#5024 part 2)
+import FolioNumber from './FolioNumber.vue'
+
+@Component({
+  components: { FolioNumber }
+})
+export default class FolioInformation extends Mixins(AuthApiMixin, CommonMixin) {
+  // Global getters
+  @Getter getFolioNumber!: string
+  @Getter isCorrectionFiling!: boolean
+  @Getter isAlterationFiling!: boolean
+  @Getter getOriginalIA!: IncorporationFilingIF
+  @Getter getSnapshotFolioNumber!: string
+
+  // Global setters
+  @Action setFolioNumber!: ActionBindingIF
+  @Action setEditingFolioNumber!: ActionBindingIF
+  @Action setValidComponent!: ActionBindingIF
+
+  /** Whether to show invalid section styling. */
+  @Prop({ default: false })
+  readonly invalidSection!: boolean
+
+  /** The original folio number dependant on filing type. */
+  private get originalFolioNumber (): string {
+    if (this.isCorrectionFiling) return this.getOriginalIA.header.folioNumber
+    if (this.isAlterationFiling) return this.getSnapshotFolioNumber
+    return null
+  }
+
+  /** On new folio number, updates auth db and store. */
+  async onNewFolioNumber (val: string): Promise<void> {
+    try {
+      if (this.isAlterationFiling) await this.updateFolioNumber(val)
+      this.setFolioNumber(val)
+    } catch (error) {
+      console.log('Update folio number error =', error) // eslint-disable-line no-console
+      this.$root.$emit('update-error-event', 'Failed to update Folio Number')
+      // reset folio number to previous value
+      const prev = this.getFolioNumber
+      // toggle for reactivity
+      this.setFolioNumber(undefined)
+      Vue.nextTick(() => this.setFolioNumber(prev))
```

Comment:
Could you just elaborate on this a little. In an error state we are rolling back here, more or less?

Reply:
Yes, exactly ("reset folio number to previous value").
But in order for all code that watches FN in the store (either via a `@Watch` or simple reactivity), I need to toggle it here, which is why I first set it to undefined, and then in a future timeslice, restore it.

Reply:
Do you want me to update the code comments? (I don't mind)

Reply:
That's ok! It's not overly complex, the code is pretty self explanatory, due to your param name 👍